### DISCUSSION TITLE
Not returning to parent folder

### DISCUSSION THREAD

**Discussion by OofMon2301** [SilentVoid13/Templater#1147] [Help] [answered]
I've had no troubles using Templater for the last year or so, but since the start of the new Semester for my university, I decided to reorganise my directories to make it much clearer for notes and to track the topics. Unfortunately, changing the folders and adding extra parent folders makes Templater hard to choose the right folder that I want it to output. I'm not very good at Templater so I am not sure what to do for this new directory.

Here is the current directory:
<img width="1280" alt="wrong directory" src="https://github.com/SilentVoid13/Templater/assets/114561394/1aec19aa-943e-40c3-92ba-a58b15984262">

And here is the template file:
<img width="475" alt="templater file" src="https://github.com/SilentVoid13/Templater/assets/114561394/cb0a2a4d-7150-481a-ab7a-e3ba87ecb555">

I want it to output the course code instead of the parent folder like before (such as MATH1051, ENGG1100, etc.)
I'm not very well versed with Templater so I thought it'd be alright to come here to ask for some help. Thank you so much :>

**Comment by Zachatoo** [comment, marked as answer]
So instead of getting the parent folder, you want to go two levels up, correct?

Instead of
```js
<% tp.file.folder(true).split('/').pop() %>
```
you can do
```js
<% tp.file.folder(true).split('/')[2] %>
```

The `[2]` will get the third folder from the root of the vault, since JavaScript arrays have [zero based indices](https://en.wikipedia.org/wiki/Zero-based_numbering). You can do whatever level folder you want, just minus one to get the level you want.

**Comment by OofMon2301** [comment]
> So instead of getting the parent folder, you want to go two levels up, correct?

Yes, this is correct. Thank you for the help. Does this `[2]` also work for this?

```c
<% this.app.workspace.getActiveFile().parent.name %>
```

**Reply by Zachatoo** [reply]
I've never seen that before, what does that do? I'm guessing it would not work in this case, since `[x]` syntax works for arrays, and I'm not seeing an array there, looks like it's returning a single file.

**Reply by OofMon2301** [reply]
> I've never seen that before, what does that do? I'm guessing it would not work in this case, since `[x]` syntax works for arrays, and I'm not seeing an array there, looks like it's returning a single file.

It uses the current tFile and then output the parent folder's name. 

It's explained in #1013 in case my explanation was not very clear.

**Reply by Zachatoo** [reply]
Based on that issue you linked, sounds like it did the same thing, so you can just replace it with my code
